client/src/partials/admin/creditor_groups/creditor_groups.js

```diff
@@ -0,0 +1,116 @@
+angular.module('bhima.controllers')
+.controller('CreditorGroupController', CreditorGroupController);
+
+CreditorGroupController.$inject = [
+  '$state', 'CreditorGroupService', 'AccountService',
+  'NotifyService', 'ModalService'
+];
+
+/**
+ * This controller is responsible for loading creditor groups and providing basic
+ * crud operations for creditor groups
+ *
+ * @module admin/creditor_groups
+ */
+function CreditorGroupController($state, CreditorGroup, Accounts, Notify, Modal) {
+  var vm = this;
+
+  var uuid = $state.params.uuid;
+
+  // global variables
+  vm.bundle = {};
+  vm.reload = { reload: true };
+  vm.isDefaultState = ($state.current.name === 'creditorGroups');
+  vm.isUpdateState = ($state.current.name === 'creditorGroups.update' && uuid);
+  vm.isCreateState = ($state.current.name === 'creditorGroups.create');
+
+  // expose to public
+  vm.submit = submit;
+  vm.deleteGroup = deleteGroup;
+
+  // load the list state
+  loadListState();
+
+  // load the detail of the creditor group
+  loadDetails();
+
+
+  // load creditor groups
+  CreditorGroup.read(null, { detailed : 1 })
+  .then(function (list) {
+    vm.creditorGroupList = list;
+  })
+  .catch(Notify.handleError);
+
+  // load accounts
+  Accounts.read(null, { detailed: 1 })
+  .then(function (list) {
+    vm.accounts = list;
+  })
+  .catch(Notify.handleError);
+
+  /**
+   * @function loadListState
+   * set the app to creditorGroup.list
+   */
+  function loadListState() {
+    if (!vm.isDefaultState) { return; }
```

Comment:
Your code currently does this:
```js
loadListState();

// ... code ...

function loadListState() {
  if (!vm.isDefaultState) { return; }

  // etc ...
}
```

It would be better to never call `loadListState()` if we are not is in the defaultState.  To me, this is a much nicer structure:
```js
if (vm.isDefaultState) {
  loadListState();  // no need to check for !vm.isDefaultState in the function.
}
```

It makes your intentions clear - everytime you call `loadListState()` you will definitely load the list state.